### PR TITLE
Use postcss-safe-parser

### PR DESCRIPTION
**Pull request by borgitas21**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [ ] Added/Updated related documentation.
- [ ] Added/Updated related tests.

## Short description of the change(s)

Use postcss-safe-parser for fault tolerance #1580

### CODE REVIEW

File: packages/parser-css/tests/integration-tests.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    const code = '.foo { color: #fff } a {';
+
+/*     await t.context.engine.emitAsync('fetch::end::css', {
+        resource: 'styles.css',
+        response: {
+            body: { content: code },
+            mediaType: 'text/css'
+        }
+    }); */
+
```
Comment (borgitas21) [inline packages/parser-css/tests/integration-tests.ts]: @antross I'm not sure how to do a real integration test without mocking the `fetch::end::css`. Could you guide me a bit?

Comment (antross) [reply packages/parser-css/tests/integration-tests.ts]: Mocking `fetch::end::css` is fine. That's consistent with what we do for similar tests in `parser-html` and `parser-javascript`. My main interest is having some coverage that actually runs `postcss` with both well-formed and malformed input which this would still do.

Comment (borgitas21) [reply packages/parser-css/tests/integration-tests.ts]: 👍

File: packages/hint-compat-api/package.json [hunk 2]
```diff
@@ -24,6 +24,7 @@
     "npm-run-all": "^4.1.5",
     "nyc": "^13.1.0",
     "postcss": "^7.0.13",
+    "postcss-safe-parser": "^4.0.1",
```
Comment (molant) [inline packages/hint-compat-api/package.json]: Why do we need `postcss` and `postcss-safe-parser` in the `compat-api` hint? Isn't all the work done in the parser?

**Comment by molant** [comment]
@borgitas21 can you please rebase with the latest in master and fix the conflicts?

File: packages/hint-compat-api/package.json (hunk 2 above)
Comment (borgitas21) [reply packages/hint-compat-api/package.json]: Yes, you are right, but since we had to add postcss to this hint, I thought we need to manage it in the same way. Maybe postcss is only useful for typing purposes in the compat-api hint. Is that correct?

**Comment by borgitas21** [comment]
> @borgitas21 can you please rebase with the latest in master and fix the conflicts?
@molant Done!

**Comment by antross** [comment]
@borgitas21 We've been working on migrating our tests to avoid `.serial` so I wanted to make some more significant changes. I also previously missed that there was a "basic-tests" file already along the lines of the "integration" tests I was looking for (just needing a malformed case). Additionally I wanted to ensure we didn't pollute our "public" types in `types.ts` with test-only types.

TL;DR I took the core of this PR and opened #1757 with the set of tests trimmed back a bit and refactored to run in parallel. Thanks for getting this started!
